### DISCUSSION TITLE
Requiring validity of one or other tag on the same field

### DISCUSSION THREAD

**Discussion by ysomad** [go-playground/validator#913] [Q&A] [answered]
I have `login` field in request struct and I want to validate it as email or username. Can I do that using `go-playground/validator` pkg?

**Comment by zemzale** [comment, marked as answer]
Depends on what are the requirements for the username. But lets say its supposed to be only alphanumerics it would looks something like
```go
type Req struct {
    Login string `validate:"required,email|alphanum"`
}
```

You can read more about other validations in docs 
https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Using_Validator_Tags

**Reply by ysomad** [reply]
Thank you! I'll check.
